server/cluster/unsafe_recovery_controller.go

```diff
@@ -615,13 +625,22 @@ func (u *unsafeRecoveryController) recordAffectedRegion(region *metapb.Region) {
 	}
 }
 
+func (u *unsafeRecoveryController) isFailed(peer *metapb.Peer) bool {
+	_, isFailed := u.failedStores[peer.StoreId]
+	_, isLive := u.storeReports[peer.StoreId]
```

Comment:
Do we need to wait for at least a round of store heartbeat?

Reply:
can't avoid that for all the corner cases, adds caveat in the flag comment